### PR TITLE
Strict null checking snippetsService.ts

### PR DESCRIPTION
**Pull request by mjbvz**
- Add conditionals around a few places that were accessing potentially null or undefined values
- Annotate which values can be null or undefined

### CODE REVIEW

File: src/vs/workbench/parts/snippets/electron-browser/snippetsService.ts [hunk 1]
```diff
@@ -203,9 +210,12 @@ class SnippetsService implements ISnippetsService {
 						continue;
 					}
 
-					if (this._files.has(validContribution.location.toString())) {
-						this._files.get(validContribution.location.toString()).defaultScopes.push(validContribution.language);
-
+					const resource = validContribution.location.toString();
+					if (this._files.has(resource)) {
+						const file = this._files.get(resource);
+						if (file.defaultScopes) {
```
Comment (mjbvz) [inline src/vs/workbench/parts/snippets/electron-browser/snippetsService.ts]: This was the one change I was unsure about. Not sure what we are supposed to do if we get here and `file.defaultScopes` is undefined. Don't do anything? Create a new array?

**Comment by mjbvz** [comment]
@jrieken Can you please take a quick look when you have a chance

File: src/vs/workbench/parts/snippets/electron-browser/snippetsService.ts (hunk 1 above)
Comment (jrieken) [reply src/vs/workbench/parts/snippets/electron-browser/snippetsService.ts]: Yeah, a new array should be created. On the other hand this is a rare corner case and only happens when different snippet contributions point to the same file.. We can also nuke that if and always let the last one win...

**Review by jrieken — APPROVED** [review]
Looking good. I'd play safe and init the `defaultScopes` in that one case

**Pull request merged**